doc/rtd/explanation/kernel-cmdline.rst

```diff
@@ -3,20 +3,44 @@
 Kernel command line
 *******************
 
+Providing configuration data via the kernel command line is somewhat of a last
+resort, as it often requires modifying the boot loader to work. Despite the
```

Comment:
I agree about the difficulty of kernel commandline injection when launching a VM.

However, I think that there is value in encouraging users to use dedicated datasource 
methods when possible instead of this user interface.

How about this:
```
 Providing configuration data via the kernel command line is somewhat of a last
-resort, as it often requires modifying the boot loader to work. Despite the
-limitations of using the kernel commandline, cloud-init supports some
-use-cases.
+resort, since this method supports only a subset of the
+:ref:`cloud config<user_data_cloud_config>` format, and many datasources
+do not support injecting kernel commandline arguments.
+
+Despite the limitations of using the kernel commandline, cloud-init supports
+some use-cases.
 
```

Reply:
Works for me!